File: ZeroMQ/testing/02req/Ping.py
import zmq, json
context = zmq.Context()
socket = context.socket(zmq.REQ)
socket.connect("tcp://localhost:19412")
dic = {'function':'Ping'}
print(dic)
socket.send_string(json.dumps(dic))
respond = socket.recv()
try:
    result = json.loads(respond.decode())
    if result:
        print(result.keys())
        res = result.get('res', 2)
        print(res)
except Exception as e:
    logging.error(e, exc_info=True)
    logging.error("respond " + respond)

File: ZeroMQ/testing/02req/Ping.js
var zmq = require('zeromq')

var socket = zmq.socket('req')
socket.connect('tcp://localhost:19412')
socket.on('message', function(msg) {
  console.log(msg.toString())
})
var dic = {'function':'Ping'}
socket.send(JSON.stringify(dic));
